### PR TITLE
Engine coupling

### PR DESCRIPTION
**Pull request by jaytaph**
Coupling the gosub_engine as a dependency to the browser. It should render the loaded html instead of just raw displaying it.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -33,6 +33,11 @@ jobs:
           key: ${{ matrix.os }}-cargo-registry-${{ matrix.rust_version }}-${{ hashFiles('**/Cargo.lock') }}
           restore-keys: |
             ${{ matrix.os }}-cargo-registry-${{ matrix.rust_version }}-
+      - name: Add Gosub crate registry
+        run: |
+          echo "[registries.gosub]" >> ~/.cargo/config.toml
+          echo "index = \"sparse+https://registry.gosub.io/\"" >> ~/.cargo/config.toml
+          echo "" >> ~/.cargo/config.toml
```
Comment (Sharktheone) [inline .github/workflows/ci.yaml]: Can't you just directly specify the registry (url) in the Cargo.toml?

**Review by Sharktheone — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (jaytaph) [reply .github/workflows/ci.yaml]: i'm not sure actually..  i can try

Comment (jaytaph) [reply .github/workflows/ci.yaml]: looks like you can't. You can only reference direct git repositories, not registry indexes.

Comment (Sharktheone) [reply .github/workflows/ci.yaml]: Ah, but you can move the `.cargo/config.toml` into the github repository, then it is way easier to run the browser

**Review by Sharktheone — APPROVED** [review]
I think the code will be a lot cleaner with the new `EngineInstance` but without this looks good.

**Pull request merged**